sdk/core/azure-core-amqp/src/models/amqp_value.cpp

```diff
@@ -0,0 +1,848 @@
+    }
+    return value;
+  }
+
+  AmqpValue::operator std::int8_t() const
+  {
+    char value;
+    if (amqpvalue_get_byte(m_value.get(), &value) != 0)
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator char() const
+  {
+    char value;
+    if (amqpvalue_get_byte(m_value.get(), &value) != 0)
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator uint16_t() const
+  {
+    uint16_t value;
+    if (amqpvalue_get_ushort(m_value.get(), &value) != 0)
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator int16_t() const
+  {
+    int16_t value;
+    if (amqpvalue_get_short(m_value.get(), &value) != 0)
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator std::uint32_t() const
+  {
+    std::uint32_t value;
+    if (amqpvalue_get_uint(m_value.get(), &value) != 0)
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator std::int32_t() const
+  {
+    int32_t value;
+    if (amqpvalue_get_int(m_value.get(), &value) != 0)
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator uint64_t() const
+  {
+    uint64_t value;
+    if (amqpvalue_get_ulong(m_value.get(), &value) != 0)
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator int64_t() const
+  {
+    int64_t value;
+    if (amqpvalue_get_long(m_value.get(), &value) != 0)
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator float() const
+  {
+    float value;
+    if (amqpvalue_get_float(m_value.get(), &value))
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator double() const
+  {
+    double value;
+    if (amqpvalue_get_double(m_value.get(), &value))
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator char32_t() const
+  {
+    std::uint32_t value;
+    if (amqpvalue_get_char(m_value.get(), &value))
+    {
+      throw std::runtime_error("Could not get character.");
+    }
+    return value;
+  }
+
+  AmqpValue::operator std::string() const
+  {
+    const char* value;
+    if (amqpvalue_get_string(m_value.get(), &value))
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator Azure::Core::Uuid() const
+  {
+    uuid value;
+    if (amqpvalue_get_uuid(m_value.get(), &value))
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    std::array<uint8_t, 16> uuid;
```

Comment:
Where is this `uuid` type defined in/coming from for `value`? For a second, I got confused by the local named `uuid` later down.

Reply:
It's a uamqp type.